bpdm-pool/src/main/kotlin/org/eclipse/tractusx/bpdm/pool/repository/LegalEntityRepository.kt

```diff
@@ -37,6 +38,10 @@ interface LegalEntityRepository : PagingAndSortingRepository<LegalEntity, Long>,
 
     fun findByUpdatedAtAfter(updatedAt: Instant, pageable: Pageable): Page<LegalEntity>
 
+    @Query("SELECT p FROM LegalEntity p WHERE LOWER(p.legalName.value) LIKE LOWER(CONCAT('%', :value, '%')) " +
+            "ORDER BY CASE WHEN LOWER(p.legalName.value) = LOWER(:value) THEN 0 WHEN LENGTH(p.legalName.value) <= LENGTH(:value) THEN 1 ELSE 2 END, LENGTH(p.legalName.value) DESC")
```

Comment:
What is the reason for having the case logic? I think the LIKE functionality can not result in the length of resulting name being shorter than the search string, correct? So just ordering by Length ascending should be fine

Reply:
I have followed the approach just with ordering by Length and removed case logic.